cli/command/container/run.go

```diff
@@ -150,7 +150,12 @@ func runContainer(ctx context.Context, dockerCli command.Cli, runOpts *runOption
 	}
 	if runOpts.sigProxy {
 		sigc := notifyAllSignals()
-		go ForwardAllSignals(ctx, apiClient, containerID, sigc)
+		// since the ForwardAllSignals already registers its own signal handler
```

Comment:
`ForwardAllSignals` doesn't set up it's own signal handler though, from looking at it's implementation? The actual signal handling setup is in `notifyAllSignals`, so maybe we can adjust this comment to be clearer.
Something like 
```
// since we're explicitly setting up signal handling here, and the daemon will
// get notified independently of the clients ctx cancellation, we use this context
// but without cancellation to avoid ForwardAllSignals from returning
// before all signals are forwarded.
```